std/experimental/color/rgb.d

```diff
@@ -0,0 +1,446 @@
+// Written in the D programming language.
+
+/**
+    This module implements the RGB _color type.
+
+    Authors:    Manu Evans
+    Copyright:  Copyright (c) 2015, Manu Evans.
+    License:    $(WEB boost.org/LICENSE_1_0.txt, Boost License 1.0)
+    Source:     $(PHOBOSSRC std/experimental/color/rgb.d)
+*/
+module std.experimental.color.rgb;
+
+import std.experimental.color;
+import std.experimental.color.conv;
+
+import std.traits: isInstanceOf, isNumeric, isIntegral, isFloatingPoint, isSigned, isSomeChar, Unqual;
+import std.typetuple: TypeTuple;
+import std.typecons: tuple;
+
+@safe: pure: nothrow: @nogc:
+
+enum isValidComponentType(T) = isIntegral!T || isFloatingPoint!T;
+
+
+/**
+Detect whether $(D T) is an RGB color.
+*/
+enum isRGB(T) = isInstanceOf!(RGB, T);
+
+
+// DEBATE: which should it be?
+template defaultAlpha(T)
+{
+/+
+    enum defaultAlpha = isFloatingPoint!T ? T(1) : T.max;
++/
+    enum defaultAlpha = T(0);
+}
+
+
+/**
+Enum of RGB color spaces.
+*/
+enum RGBColorSpace
+{
+    /** sRGB, HDTV (ITU-R BT.709) */
+    sRGB,
+    /** sRGB with gamma 2.2 */
+    sRGB_Gamma2_2,
+
+    // custom color space will disable automatic color spoace conversions
+    custom = -1
+}
+
+
+/**
+An RGB color, parameterised with components, component type, and color space specification.
+
+Params: components_ = Components that shall be available. Struct is populated with components in the order specified.
+                      Valid components are:
+                        "r" = red
+                        "g" = green
+                        "b" = blue
+                        "a" = alpha
+                        "l" = luminance
+                        "x" = placeholder/padding (no significant value)
+        ComponentType_ = Type for the color channels. May be a basic integer or floating point type.
+        linear_ = Color is stored with linear luminance.
+        colorSpace_ = Color will be within the specified color space.
+*/
+struct RGB(string components_, ComponentType_, bool linear_ = false, RGBColorSpace colorSpace_ = RGBColorSpace.sRGB) if(isValidComponentType!ComponentType_)
+{
+@safe: pure: nothrow: @nogc:
+
+    // RGB colors may only contain components 'rgb', or 'l' (luminance)
+    // They may also optionally contain an 'a' (alpha) component, and 'x' (unused) components
+    static assert(allIn!("rgblax", components), "Invalid Color component '"d ~ notIn!("rgblax", components) ~ "'. RGB colors may only contain components: r, g, b, l, a, x"d);
+    static assert(anyIn!("rgbal", components), "RGB colors must contain at least one component of r, g, b, l, a.");
+    static assert(!canFind!(components, 'l') || !anyIn!("rgb", components), "RGB colors may not contain rgb AND luminance components together.");
+
+    // create members for some useful information
+    /** Type of the color components. */
+    alias ComponentType = ComponentType_;
```

Comment:
this is redudant, just rename the struct template parameter to CompoentType and remove this line